ui/src/Grid/Message.elm

```diff
@@ -0,0 +1,7 @@
+module Grid.Message exposing (..)
+
+import Grid.Player exposing (..)
+
+type Msg = PlayerMoved PlayerId Position Orientation
+         | PlayerDied PlayerId
+         | NoOp
```

Comment:
this is `elm` code. How do we share `haskell` code here ? or you are talking about something else ?
Also, not all msgs are required in client side like `GameEnded`..